war/src/main/java/io/lumeer/engine/controller/search/QuerySuggester.java

```diff
@@ -105,41 +104,56 @@
    List<SearchSuggestion> suggestAttributes(String text, int limit) {
       String[] parts = text.split("\\.", 2);
       if (parts.length < 2) {
-         return suggestCollectionsForAttribute(text, limit); // TODO look for attributes of all collections instead
+         return suggestAttributesForAllCollections(text, limit);
       }
 
       String collectionName = parts[0];
       String attributePart = parts[1];
 
       try {
          CollectionMetadata collectionMetadata = collectionMetadataFacade.getCollectionMetadata(collectionMetadataFacade.getInternalCollectionName(collectionName));
-         return collectionMetadata.getAttributes().stream()
-                                  .filter(attribute -> attribute.getName().toLowerCase().contains(attributePart.toLowerCase()))
-                                  .map(attribute -> convertAttribute(collectionName, attribute))
-                                  .collect(Collectors.toList()); // TODO do not ignore child attributes
+         return filterAttributes(collectionMetadata, attributePart, limit);
       } catch (UserCollectionNotFoundException e) {
          return Collections.emptyList();
       }
    }
 
+   private List<SearchSuggestion> suggestAttributesForAllCollections(String attributePart, int limit) {
+      String metadataCollection = collectionMetadataFacade.metadataCollection();
+      List<DataDocument> collectionsRaw = userDataStorage.search(metadataCollection, null, null);
```

Comment:
Isn't it possible to get attributes directly instead of looping through all metadata collections?